pandas/core/frame.py

```diff
@@ -5655,12 +5664,21 @@ def reset_index(
         col_fill : object, default ''
             If the columns have multiple levels, determines how the other
             levels are named. If None then the index name is repeated.
+        allow_duplicates : bool, optional, default lib.no_default
+            Allow duplicate column labels to be created.
+
+            .. versionadded:: 1.4.0
 
         Returns
         -------
         DataFrame or None
             DataFrame with the new index or None if ``inplace=True``.
 
+        Raises
+        ------
+        ValueError
+            If allow_duplicates and not self.flags.allows_duplicate_labels.
```

Comment:
this doesn't make any sense

Reply:
I am just reporting what insert does. No more, no less. If it doesn't make any sense then that is a whole other issue and is nothing to do with this PR.

Reply:
Perhaps it's time to take another look at this.

The "policy" is set in DataFrame.insert. If that policy is wrong, then that is another issue and is _not_ the subject of this issue or PR. 

Other reviewers seem to agree with that policy, so I can't see it being changed.

In any case, I have factored the behavior all the way down to DataFrame.Insert, so if it did ever need changing then it could be done in one place.